src/game/game_demo.c

```diff
@@ -15,11 +14,67 @@
 
 #include <stdbool.h>
 #include <stddef.h>
+#include <stdint.h>
 
 static int32_t m_DemoLevel = -1;
 static uint32_t *m_DemoPtr = NULL;
 
-int32_t StartDemo(void)
+static void Game_Demo_LoadLaraPos(void);
+
+static void Game_Demo_LoadLaraPos(void)
+{
+    m_DemoPtr = g_DemoData;
+    ITEM_INFO *item = g_LaraItem;
+    item->pos.x = *m_DemoPtr++;
+    item->pos.y = *m_DemoPtr++;
+    item->pos.z = *m_DemoPtr++;
+    item->pos.x_rot = *m_DemoPtr++;
+    item->pos.y_rot = *m_DemoPtr++;
+    item->pos.z_rot = *m_DemoPtr++;
+    int16_t room_num = *m_DemoPtr++;
+
+    if (item->room_number != room_num) {
+        Item_NewRoom(g_Lara.item_number, room_num);
+    }
+
+    FLOOR_INFO *floor =
+        Room_GetFloor(item->pos.x, item->pos.y, item->pos.z, &room_num);
+    item->floor = Room_GetHeight(floor, item->pos.x, item->pos.y, item->pos.z);
+}
+
+bool Game_Demo_ProcessInput(void)
+{
+    if (m_DemoPtr >= &g_DemoData[DEMO_COUNT_MAX] || (int)*m_DemoPtr == -1) {
+        return false;
+    }
+
+    // Translate demo inputs (that use TombATI key values) to T1M inputs.
+    g_Input = (INPUT_STATE) {
+        0,
+        .forward = (bool)(*m_DemoPtr & (1 << 0)),
+        .back = (bool)(*m_DemoPtr & (1 << 1)),
+        .left = (bool)(*m_DemoPtr & (1 << 2)),
+        .right = (bool)(*m_DemoPtr & (1 << 3)),
+        .jump = (bool)(*m_DemoPtr & (1 << 4)),
+        .draw = (bool)(*m_DemoPtr & (1 << 5)),
+        .action = (bool)(*m_DemoPtr & (1 << 6)),
+        .slow = (bool)(*m_DemoPtr & (1 << 7)),
+        .option = (bool)(*m_DemoPtr & (1 << 8)),
+        .look = (bool)(*m_DemoPtr & (1 << 9)),
+        .step_left = (bool)(*m_DemoPtr & (1 << 10)),
+        .step_right = (bool)(*m_DemoPtr & (1 << 11)),
+        .roll = (bool)(*m_DemoPtr & (1 << 12)),
+        .select = (bool)(*m_DemoPtr & (1 << 20)),
+        .deselect = (bool)(*m_DemoPtr & (1 << 21)),
+        .save = (bool)(*m_DemoPtr & (1 << 22)),
+        .load = (bool)(*m_DemoPtr & (1 << 23)),
+    };
+
+    m_DemoPtr++;
+    return true;
+}
+
+void Game_Demo(void)
```

Comment:
`Game_Demo_Start` to match `Game_Start`? I guess it's not really a big deal.

Reply:
Unlike `Game_Cutscene_Start` which is similarly sliced, this routine runs the entire loop (similar to `Game_Pause`).